# Call Data

In [19]:
#clean up csv files
import pandas as pd

In [20]:
#variables for files
data = 'Data/clean_df.csv'
genre_data = 'Data/clean_genre.csv'

In [21]:
#import csvs
genre_data = pd.read_csv(genre_data)
data = pd.read_csv(data)

# Find Most Popular Genre

In [22]:
#Print genres dataframe
genre_data

Genres  Danceability (0-1)  Tempo (BPM)  \
0     21st century classical            0.284100    91.351000   
1                      432hz            0.312000   118.900933   
2                      8-bit            0.673000   192.816000   
3                 a cappella            0.516172   105.506031   
4                   abstract            0.473600   142.495400   
...                      ...                 ...          ...   
3226        zim urban groove            0.553000   134.995000   
3227                    zolo            0.533837   126.765194   
3228                    zouk            0.641889   105.848889   
3229            zurich indie            0.705667    91.278000   
3230                  zydeco            0.625963   130.137242   

      Popularity (0-100) Key   Mode  
0               6.600000   E  Major  
1              41.200000   B  Major  
2               0.000000   F  Major  
3              39.086248   G  Major  
4              43.800000   F  Major  
...                  ...  ..    ...  
3226            9.000000   G  Major  
3227           31.108254   A  Major  
3228           32.555556   G  Major  
3229            0.000000   G  Minor  
3230           27.703810   A  Major  

[3231 rows x 6 columns]

In [52]:
#Make genre column into string so all data is the same type
genre_data['Genres'] = genre_data['Genres'].apply(str)

#Sort Popularity in descending order to confirm most popular genre.
sorted_genre = genre_data.sort_values("Popularity (0-100)", ascending=False)
sorted_genre

Genres  Danceability (0-1)  Tempo (BPM)  Popularity (0-100)  \
564   chinese electropop               0.660      142.018                79.0   
1767  korean mask singer               0.658      128.909                78.0   
995        dutch rap pop               0.767      169.990                77.0   
3212                yaoi               0.603      135.028                77.0   
944              dong-yo               0.829      115.056                76.0   
...                  ...                 ...          ...                 ...   
2089           neo-kraut               0.705      123.993                 0.0   
573             chip hop               0.780       77.060                 0.0   
2283       pagode baiano               0.642      211.968                 0.0   
256               batida               0.861      119.984                 0.0   
823       dark psytrance               0.725      147.006                 0.0   

     Key   Mode  
564   Db  Minor  
1767   A  Minor  
995    F  Major  
3212  Ab  Minor  
944   Ab  Major  
...   ..    ...  
2089   A  Major  
573   Eb  Major  
2283   E  Major  
256    B  Major  
823    B  Minor  

[3231 rows x 6 columns]

With the cleaned and sorted data, we find that the most popular genre is Chinese Electropop.

In [59]:
#Sort Popularity in descending order to confirm most popular artist.
sorted_data = data.sort_values("Popularity (0-100)", ascending=False)
sorted_data = sorted_data.dropna()
sorted_data

Artists  Danceability (0-1)  \
20062                                      Olivia Rodrigo               0.585   
19862                               24kGoldn', 'iann dior               0.700   
19866                                       Ariana Grande               0.737   
19886                            Bad Bunny', 'Jhay Cortez               0.731   
19976                                             KAROL G               0.863   
...                                                   ...                 ...   
63064                                      Tarun Banerjee               0.530   
63063                                    Georges Derveaux               0.209   
63062                                    Эрнест Хемингуэй               0.800   
63061   Hector Berlioz', 'Concertgebouworkest', 'Eduar...               0.276   
174388                                          Roger Fly               0.677   

        Duration (seconds) Explicit Key   Mode  \
20062              242.014      Yes  Bb  Major   
19862              140.526      Yes   G  Minor   
19866              172.325      Yes   C  Major   
19886              205.090      Yes   E  Minor   
19976              178.947      Yes  Db  Minor   
...                    ...      ...  ..    ...   
63064              199.227       No   B  Minor   
63063              162.623       No   D  Major   
63062               95.700      Yes  Gb  Major   
63061              370.787       No   A  Major   
174388             197.710       No   G  Major   

                                               Song Title  Popularity (0-100)  \
20062                                     drivers license                 100   
19862                              Mood (feat. iann dior)                  96   
19866                                           positions                  96   
19886                                              DÁKITI                  95   
19976                                             BICHOTA                  95   
...                                                   ...                 ...   
63064                                        Phul Nebe Go                   0   
63063                                            Habanera                   0   
63062                          Часть 44.3 - Старик и море                   0   
63061   Symphonie fantastique, Op. 14, H 48: 2. Un bal...                   0   
174388                                     Improvisations                   0   

        Tempo (BPM)  Release Year  
20062       143.874          2021  
19862        90.989          2020  
19866       144.015          2020  
19886       109.928          2020  
19976       163.908          2020  
...             ...           ...  
63064        79.712          1952  
63063       139.475          1952  
63062        42.490          1952  
63061        89.437          1952  
174388      112.208          2020  

[174389 rows x 10 columns]

# Find Most Popular Artist

Currently Olivia Rodrigo is most popular with her song Drivers License. 

In [63]:
#To find out more about Olivia Rodrigo, we use loc to gather data regarding only her.
olivia_data =  sorted_data.loc[sorted_data["Artists"] == "Olivia Rodrigo", [
    "Artists", "Song Title", "Popularity (0-100)"]]
olivia_data

Artists                                         Song Title  \
20062   Olivia Rodrigo                                    drivers license   
19732   Olivia Rodrigo  All I Want - From "High School Musical: The Mu...   
110088  Olivia Rodrigo                                         All I Want   

        Popularity (0-100)  
20062                  100  
19732                   82  
110088                  75

All three of her songs available on Spotify are incredibly popular. 

In [64]:
#We found most popular artist based on the popularity of the song, but let us see which artist has the most songs listed.
sorted_data["Artists"].value_counts()

Tadeusz Dolega Mostowicz                                                             1281
Эрнест Хемингуэй                                                                     1175
Эрих Мария Ремарк                                                                    1062
Francisco Canaro                                                                      951
Ignacio Corsini                                                                       624
                                                                                     ... 
Stephen Sondheim', 'Len Cariou', 'Victor Garber', 'Merle Louise', 'Paul Gemignani       1
Bo Donaldson & The Heywoods                                                             1
Sunshine Anderson                                                                       1
UNKLE                                                                                   1
The Urge                                                                                1
Name: Arti

In [65]:
#To find out more about Francisco Canaro, we use loc to gather data regarding only him.
canaro =  sorted_data.loc[sorted_data["Artists"] == "Francisco Canaro", [
    "Artists", "Song Title", "Popularity (0-100)"]]
canaro

Artists                                         Song Title  \
4705   Francisco Canaro                   Soñar y Nada Mas - Remasterizado   
24208  Francisco Canaro                 Mas Solo Que Nunca - Remasterizado   
40111  Francisco Canaro              Mimosa - Instrumental (Remasterizado)   
24202  Francisco Canaro              Café de los Angelitos - Remasterizado   
21144  Francisco Canaro  La Rancherita de Santa Lucia - Instrumental (R...   
...                 ...                                                ...   
60702  Francisco Canaro    Tango Milonguero - Instrumental (Remasterizado)   
60681  Francisco Canaro  Bandoneón de mis amores - Instrumental (Remast...   
61187  Francisco Canaro                        Anoche Soñé - Remasterizado   
61185  Francisco Canaro                  Decile Que Vuelva - Remasterizado   
61168  Francisco Canaro             La Vida en 1000 Gramos - Remasterizado   

       Popularity (0-100)  
4705                   24  
24208                   4  
40111                   2  
24202                   2  
21144                   2  
...                   ...  
60702                   0  
60681                   0  
61187                   0  
61185                   0  
61168                   0  

[951 rows x 3 columns]

Tadeusz Dolega Mostowicz has the most titles on Spotify with a stunning 1,281 total. These titles, however, are snippets of audiobooks and not songs. The first music artist with the greatest amount of pieces is Francisco Canaro, a violinist from Uruguay. After generating a dataframe reflectig his popularity, the quantity of his titles does not display quality, his popularity far lower than Olivia Rodrigo's.

# Find the Artists Currently Most Popular per Decade

In [94]:
#Find range for years.
data.describe()

#create new call-name for dataframe
decade_data = data

#Minimum year is 1920, with max being 2021. Now we set our bins.
bins = [1920, 1929, 1939, 1949, 1959, 1969, 1979, 1989, 1999, 2009, 2019]

decade = ["1920s", "1930s", "1940s", "1950s", "1960s", "1970s", "1980s", "1990s", "2000s", "2010s"]

#Using bins we create a column just for them.
decade_data["Decades"] = pd.cut(data["Release Year"], bins, labels=decade, include_lowest=True)

#Drop rows for 2020 and 2021, as these are for an incomplete decade.
decade_data.dropna()

Artists  Danceability (0-1)  Duration (seconds)  \
0                 Mamie Smith               0.598             168.333   
1       Screamin' Jay Hawkins               0.852             150.200   
2                 Mamie Smith               0.647             163.827   
3             Oscar Velazquez               0.730             422.087   
4                        Mixe               0.704             165.224   
...                       ...                 ...                 ...   
174116         Workout Music                0.734             277.493   
174118         Workout Music                0.533             287.987   
174120         Workout Music                0.764             268.120   
174124                October               0.520             226.603   
174128            DJ X-Change               0.620              30.250   

       Explicit Key   Mode                                         Song Title  \
0            No   F  Minor                           Keep A Song In Your Soul   
1            No   F  Minor                               I Put A Spell On You   
2            No   C  Major                                       Golfing Papa   
3            No   D  Major  True House Music - Xavier Santos & Carlos Gomi...   
4           Yes  Bb  Minor                                          Xuniverxe   
...         ...  ..    ...                                                ...   
174116       No   C  Minor                  RITMO (Bay Boys For Life) [Remix]   
174118       No   F  Major                           Beautiful People (Remix)   
174120       No   D  Major                                       Tusa (Remix)   
174124       No  Db  Major                      A Good Year - 2020 Remastered   
174128       No   C  Major              New Years Eve Ultimate Countdown 2020   

        Popularity (0-100)  Tempo (BPM)  Release Year Decades  
0                       12      149.976          1920   1920s  
1                        7       86.889          1920   1920s  
2                        4       97.600          1920   1920s  
3                       17      127.997          1920   1920s  
4                        2      122.076          1920   1920s  
...                    ...          ...           ...     ...  
174116                  14      128.015          2019   2010s  
174118                  14      129.943          2019   2010s  
174120                  16      127.994          2019   2010s  
174124                   5      123.976          2013   2010s  
174128                   8       93.056          2019   2010s  

[168255 rows x 11 columns]

In [96]:
#Focus on decade data, create dataframe for 1920s.
twenties =  decade_data.loc[decade_data["Decades"] == "1920s", [
    "Artists", "Song Title", "Popularity (0-100)", "Release Year", "Decades"]]
twenties.sort_values("Popularity (0-100)", ascending=False)

Artists  \
1662                           Louis Armstrong   
1462                             Benny Goodman   
1062                               Fats Waller   
1463                      Blind Willie Johnson   
1063                               Fats Waller   
...                                        ...   
20803    Hans Pfitzner', 'Staatskapelle Berlin   
20802                       Arthur Conan Doyle   
20800                       Arthur Conan Doyle   
20799                          Ignacio Corsini   
126973  Frédéric Chopin', 'Sergei Rachmaninoff   

                                               Song Title  Popularity (0-100)  \
1662                                       Mack the Knife                  51   
1462                                     Sing, Sing, Sing                  50   
1062                                    Ain't Misbehavin'                  47   
1463              Dark Was the Night, Cold Was the Ground                  46   
1063                                    Two Sleepy People                  44   
...                                                   ...                 ...   
20803   Die Rose vom Liebesgarten - Zwischenspiele: Fu...                   0   
20802        Chapter 61.2 & Chapter 62.1 - Dolina strachu                   0   
20800        Chapter 86.2 & Chapter 87.1 - Dolina strachu                   0   
20799                      Patio Olvidado - Remasterizado                   0   
126973                       Valse in E Minor, Op. posth.                   0   

        Release Year Decades  
1662            1929   1920s  
1462            1928   1920s  
1062            1926   1920s  
1463            1928   1920s  
1063            1926   1920s  
...              ...     ...  
20803           1927   1920s  
20802           1927   1920s  
20800           1927   1920s  
20799           1927   1920s  
126973          1926   1920s  

[5631 rows x 5 columns]

In [97]:
#Focus on decade data, create dataframe for 1930s.
thirties =  decade_data.loc[decade_data["Decades"] == "1930s", [
    "Artists", "Song Title", "Popularity (0-100)", "Release Year", "Decades"]]
thirties.sort_values("Popularity (0-100)", ascending=False)

Artists  \
2462                      Billie Holiday', 'Eddie Heywood   
3662                                          Harry James   
2463                                            Art Tatum   
3262                                       Robert Johnson   
3665                                           Lead Belly   
...                                                   ...   
41078   Charras Gonzalez', 'Veronica Gonzalez Day', 'A...   
41077   Charras Gonzalez', 'Veronica Gonzalez Day', 'A...   
41075                                     Maurice Jaubert   
41073                                      Κώστας Μαυρέας   
159188                           Tadeusz Dolega Mostowicz   

                                               Song Title  Popularity (0-100)  \
2462       All of Me (with Eddie Heywood & His Orchestra)                  63   
3662                          It's Been a Long, Long Time                  59   
2463                                          Tea for Two                  53   
3262                                     Cross Road Blues                  53   
3665                       Where Did You Sleep Last Night                  51   
...                                                   ...                 ...   
41078                             Que c´est triste Venise                   0   
41077                                             Soledad                   0   
41075        Part.1 (From En Crète sans les dieux) (1934)                   0   
41073                                      Για πες αλεύρι                   0   
159188  Chapter 300.2 & Chapter 301.1 - Drugie życie d...                   0   

        Release Year Decades  
2462            1933   1930s  
3662            1939   1930s  
2463            1933   1930s  
3262            1937   1930s  
3665            1939   1930s  
...              ...     ...  
41078           1934   1930s  
41077           1934   1930s  
41075           1934   1930s  
41073           1934   1930s  
159188          1936   1930s  

[10764 rows x 5 columns]

In [98]:
#Focus on decade data, create dataframe for 1940s.
forties =  decade_data.loc[decade_data["Decades"] == "1940s", [
    "Artists", "Song Title", "Popularity (0-100)", "Release Year", "Decades"]]
forties.sort_values("Popularity (0-100)", ascending=False)

Artists  \
4262    Bing Crosby', 'Ken Darby Singers', 'John Scott...   
4662                                         Judy Garland   
5262                                           Gene Autry   
5663                     Erik Satie', 'Philippe Entremont   
5662                   Bing Crosby', 'The Andrews Sisters   
...                                                   ...   
61280                                          Max Hansen   
61281                                   Ellen Gottschalch   
61282                Errol Flynn', 'Spike Jones Orchestra   
61283                                     Valdemar Davids   
159988                                         Geeta Dutt   

                                               Song Title  Popularity (0-100)  \
4262                                      White Christmas                  80   
4662               Have Yourself A Merry Little Christmas                  74   
5262    Here Comes Santa Claus (Right Down Santa Claus...                  73   
5663                                     Gymnopédie No. 1                  68   
5662                             Twelve Days Of Christmas                  68   
...                                                   ...                 ...   
61280                    Sommer i Tyrol/Rigdom har jeg ej                   0   
61281                      Sommer i Tyrol/I Salzkammergut                   0   
61282                      That's What You Jolly Well Get                   0   
61283                              Ved landsbyens gadekær                   0   
159988                         Nainon Mein Bhar Liya Neer                   0   

        Release Year Decades  
4262            1942   1940s  
4662            1944   1940s  
5262            1947   1940s  
5663            1949   1940s  
5662            1949   1940s  
...              ...     ...  
61280           1943   1940s  
61281           1943   1940s  
61282           1943   1940s  
61283           1943   1940s  
159988          1949   1940s  

[15493 rows x 5 columns]

In [99]:
#Focus on decade data, create dataframe for 1950s.
fifties =  decade_data.loc[decade_data["Decades"] == "1950s", [
    "Artists", "Song Title", "Popularity (0-100)", "Release Year", "Decades"]]
fifties.sort_values("Popularity (0-100)", ascending=False)

Artists  \
7262                                          Bobby Helms   
7662                                          Dean Martin   
7263                                        Elvis Presley   
7666                                          Chuck Berry   
7664                                          Dean Martin   
...                                                   ...   
113030                        G. M. Durrani', 'Geeta Dutt   
144920                               Αντώνης Καλυβόπουλος   
144919                                         Geeta Dutt   
144918                         Three male Vodoun drummers   
96320   Johann Sebastian Bach', 'Agnes Giebel', 'RIAS ...   

                                               Song Title  Popularity (0-100)  \
7262                                     Jingle Bell Rock                  85   
7662               Let It Snow! Let It Snow! Let It Snow!                  84   
7263                                       Blue Christmas                  79   
7666                                      Johnny B. Goode                  75   
7664                       Rudolph The Red-Nosed Reindeer                  75   
...                                                   ...                 ...   
113030                        Jab Tak Chamke Chand Sitare                   0   
144920                            Το μπουζούκι μου κρατάω                   0   
144919                                     Tu Kahan Chala                   0   
144918                  Vodoun Dance (Three Vodoun Drums)                   0   
96320   Christmas Oratorio, BWV 248, Pt. 6: No. 56, Re...                   0   

        Release Year Decades  
7262            1957   1950s  
7662            1959   1950s  
7263            1957   1950s  
7666            1959   1950s  
7664            1959   1950s  
...              ...     ...  
113030          1952   1950s  
144920          1950   1950s  
144919          1950   1950s  
144918          1950   1950s  
96320           1950   1950s  

[20000 rows x 5 columns]

In [100]:
#Focus on decade data, create dataframe for 1960s.
sixties =  decade_data.loc[decade_data["Decades"] == "1960s", [
    "Artists", "Song Title", "Popularity (0-100)", "Release Year", "Decades"]]
sixties.sort_values("Popularity (0-100)", ascending=False)

Artists  \
8662                                           Brenda Lee   
8462                                        Andy Williams   
9663                                          The Beatles   
8463                                         The Ronettes   
8262                                        Nat King Cole   
...                                                   ...   
162386  Howard Brubeck', 'The Dave Brubeck Quartet', '...   
162375  Lester Flatt', 'Earl Scruggs', 'Mother Maybell...   
162347  Camille Saint-Saëns', 'Leonard Bernstein', 'Ne...   
36795                                        Nancy Wilson   
35851                                         The Hollies   

                                               Song Title  Popularity (0-100)  \
8662                    Rockin' Around The Christmas Tree                  87   
8462             It's the Most Wonderful Time of the Year                  85   
9663                 Here Comes The Sun - Remastered 2009                  82   
8463                                          Sleigh Ride                  82   
8262          The Christmas Song (Merry Christmas To You)                  81   
...                                                   ...                 ...   
162386  Dialogue for Jazz Combo and Orchestra: I. Alle...                   4   
162375  Jimmie Brown, The Newsboy (with Mother Maybell...                   4   
162347             Carnival of the Animals: IV. Tortoises                   4   
36795              Ten Good Years - 2007 Digital Remaster                   2   
35851                      Hard Hard Year - 2003 Remaster                   1   

        Release Year Decades  
8662            1964   1960s  
8462            1963   1960s  
9663            1969   1960s  
8463            1963   1960s  
8262            1962   1960s  
...              ...     ...  
162386          1961   1960s  
162375          1961   1960s  
162347          1961   1960s  
36795           1965   1960s  
35851           1965   1960s  

[19610 rows x 5 columns]

In [101]:
#Focus on decade data, create dataframe for 1970s.
seventies =  decade_data.loc[decade_data["Decades"] == "1970s", [
    "Artists", "Song Title", "Popularity (0-100)", "Release Year", "Decades"]]
seventies.sort_values("Popularity (0-100)", ascending=False)

Artists                                         Song Title  \
11262    Fleetwood Mac                             Dreams - 2004 Remaster   
11062           Eagles                   Hotel California - 2013 Remaster   
11662            AC/DC                                    Highway to Hell   
10262       Elton John  Rocket Man (I Think It's Going To Be A Long, L...   
10662   Lynyrd Skynyrd                                 Sweet Home Alabama   
...                ...                                                ...   
37625  Tangerine Dream                   Rubycon - Advert / 2011 Remaster   
37613  Tangerine Dream                   Phaedra - Advert / 2011 Remaster   
35863   The Groundhogs       A Year In The Life - 2003 Remastered Version   
93005       Greg Quill             Seven Years of Silence - 2016 Remaster   
89921         Uncle Po                                  2000 Years - Live   

       Popularity (0-100)  Release Year Decades  
11262                  86          1977   1970s  
11062                  83          1976   1970s  
11662                  82          1979   1970s  
10262                  81          1972   1970s  
10662                  81          1974   1970s  
...                   ...           ...     ...  
37625                   3          1978   1970s  
37613                   3          1978   1970s  
35863                   3          1971   1970s  
93005                   0          1975   1970s  
89921                   0          1977   1970s  

[20118 rows x 5 columns]

In [102]:
#Focus on decade data, create dataframe for 1930s.
eighties =  decade_data.loc[decade_data["Decades"] == "1980s", [
    "Artists", "Song Title", "Popularity (0-100)", "Release Year", "Decades"]]
eighties.sort_values("Popularity (0-100)", ascending=False)

Artists  \
12462                     The Police   
11863                          AC/DC   
11862                 Paul McCartney   
12263                Michael Jackson   
12864                           a-ha   
...                              ...   
126204                       Octobre   
126206                       Octobre   
142620  Betha Sarasin', 'Markus Ganz   
90055                  Zulu Warriors   
106399                    Six Inches   

                                               Song Title  Popularity (0-100)  \
12462                               Every Breath You Take                  84   
11863                                       Back In Black                  84   
11862   Wonderful Christmastime - Edited Version / Rem...                  83   
12263                                         Billie Jean                  83   
12864                                          Take on Me                  83   
...                                                   ...                 ...   
126204            Élégante solution - Remasterisé en 2018                   1   
126206           Metropolitan Opera - Remasterisé en 2018                   1   
142620                                           Yearning                   0   
90055                                Dub In The Year 2000                   0   
106399                         2000 Light Years From Home                   0   

        Release Year Decades  
12462           1983   1980s  
11863           1980   1980s  
11862           1980   1980s  
12263           1982   1980s  
12864           1985   1980s  
...              ...     ...  
126204          1982   1980s  
126206          1982   1980s  
142620          1988   1980s  
90055           1989   1980s  
106399          1988   1980s  

[20074 rows x 5 columns]

In [103]:
#Focus on decade data, create dataframe for 1990s.
nineties =  decade_data.loc[decade_data["Decades"] == "1990s", [
    "Artists", "Song Title", "Popularity (0-100)", "Release Year", "Decades"]]
nineties.sort_values("Popularity (0-100)", ascending=False)

Artists  \
14662                          Mariah Carey   
14663   Frank Sinatra', 'B. Swanson Quartet   
13863                                 AC/DC   
14067                                R.E.M.   
14863                        Coolio', 'L.V.   
...                                     ...   
140273          Nathan Haines', 'Kenny Dope   
89899                                O-Trax   
89929     Graeme E. Pearson & The Mutineers   
89939                The Motor City Burgers   
170188                      Majority of One   

                                               Song Title  Popularity (0-100)  \
14662                     All I Want for Christmas Is You                  90   
14663   Let It Snow! Let It Snow! Let It Snow! (with T...                  84   
13863                                       Thunderstruck                  82   
14067                                  Losing My Religion                  81   
14863                                  Gangsta's Paradise                  81   
...                                                   ...                 ...   
140273                         Believe - Kenny Dope Remix                   0   
89899                                       The Year 2000                   0   
89929                                       The Year 2000                   0   
89939                                     Year 2000 Snafu                   0   
170188                                             Palace                   0   

        Release Year Decades  
14662           1994   1990s  
14663           1994   1990s  
13863           1990   1990s  
14067           1991   1990s  
14863           1995   1990s  
...              ...     ...  
140273          1996   1990s  
89899           1999   1990s  
89929           1996   1990s  
89939           1999   1990s  
170188          1993   1990s  

[20328 rows x 5 columns]

In [104]:
#Focus on decade data, create dataframe for 2000s.
two_thousands =  decade_data.loc[decade_data["Decades"] == "2000s", [
    "Artists", "Song Title", "Popularity (0-100)", "Release Year", "Decades"]]
two_thousands.sort_values("Popularity (0-100)", ascending=False)

Artists  \
15862                            Coldplay   
17462                       Mother Mother   
15868                         Linkin Park   
16263                 Eminem', 'Nate Dogg   
16265                            Coldplay   
...                                   ...   
91653   Cliff Bennett & The Rebel Rousers   
55735                            The Womb   
91649   Cliff Bennett & The Rebel Rousers   
140183                Tiësto', 'Maxi Jazz   
172648                          Mike Hype   

                                               Song Title  Popularity (0-100)  \
15862                                              Yellow                  85   
17462                                             Hayloft                  85   
15868                                          In the End                  83   
16263                                    'Till I Collapse                  83   
16265                                       The Scientist                  83   
...                                                   ...                 ...   
91653   Said I Weren't Gonna Tell Nobody - Mono; 2009 ...                   0   
55735                                   Five Years - Demo                   0   
91649    I'll Take Good Care of You - Mono; 2009 Remaster                   0   
140183             Dance 4 life [ASOT 281] - Original Mix                   0   
172648                                      Summer Nights                   0   

        Release Year Decades  
15862           2000   2000s  
17462           2008   2000s  
15868           2000   2000s  
16263           2002   2000s  
16265           2002   2000s  
...              ...     ...  
91653           2009   2000s  
55735           2007   2000s  
91649           2009   2000s  
140183          2006   2000s  
172648          2000   2000s  

[13782 rows x 5 columns]

In [105]:
#Focus on decade data, create dataframe for 2010s.
twenty_tens =  decade_data.loc[decade_data["Decades"] == "2010s", [
    "Artists", "Song Title", "Popularity (0-100)", "Release Year", "Decades"]]
twenty_tens.sort_values("Popularity (0-100)", ascending=False)

Artists                                   Song Title  \
19670             Harry Styles                             Watermelon Sugar   
19470                      Sia                                      Snowman   
19672            Lewis Capaldi                            Someone You Loved   
19780             Topic', 'A7S                                  Breaking Me   
18462        The Neighbourhood                              Sweater Weather   
...                        ...                                          ...   
140682                Palaraga                            Avoiding the Void   
140680       Matt Fax', 'Dezza                                  Sweet Dream   
140678         Simone Lo Porto                             Fiamma su fiamma   
140676  Eleven Fly', 'March 13                                    Day Dream   
107083                Nemystic  Every Body Looks for Something - Remastered   

        Popularity (0-100)  Release Year Decades  
19670                   91          2019   2010s  
19470                   91          2018   2010s  
19672                   90          2019   2010s  
19780                   89          2019   2010s  
18462                   89          2013   2010s  
...                    ...           ...     ...  
140682                   0          2018   2010s  
140680                   0          2018   2010s  
140678                   0          2018   2010s  
140676                   0          2018   2010s  
107083                   0          2013   2010s  

[22455 rows x 5 columns]